File: problem10.java
/*
The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.
*/
import java.util.ArrayList;
import java.util.Iterator;
public class problem10{
      public static void main(String args[]){
            int count = 0;
            int number = 2;
            long result = 0;
            int lastPrime = -1;
            ArrayList<Integer> primeList = new ArrayList<Integer>();
            while(number < 2_000_000){
                  Iterator<Integer> itI = primeList.iterator();
                  boolean isPrime = true;
                  while(itI.hasNext() ){
                        int prime = itI.next();
                        if(number%prime==0){
                              isPrime =false;
                        }
                  }
                  if(count%200==0){
                        System.out.println(number+":"+lastPrime);
                  }
                  if(isPrime){
                        lastPrime = number;
                        primeList.add(number);
                        count++;
                        result += number;
                  }
                  number++;
            }
            System.out.println(result);
      }
}

File: problem10.c
#include <stdio.h>
#include <stdbool.h>
#define _ARRAYSIZE 250000
int main(){
      int primes[_ARRAYSIZE];
      int i;
      for(i = 0; i < _ARRAYSIZE; i++){
            primes[i] = -1;
      }
      int n=2;
      int index=0;
      while(n< 2000000){
            int j;
            bool isPrime = true;
            
            for(j=0; j<index;j++){
                  //printf("checking %d\n",j);
                  int factor = primes[j];
                  if(n%factor==0){
                        isPrime = false;
                        break;
                  }
            }
            if(isPrime){
            printf("%d\n",n);
                  primes[index++]=n;
            }
            n++;
      }
      
      int j;
      long sum=0;
      for(j=0; j<index;j++){
            sum+=(long)primes[j];
      }
      printf("The sum of primes up to 2 million is %ld.", sum);
     
return 0;
}
